src/App.jsx

```diff
@@ -14,33 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortByLength = 'length';
+const sortByABC = 'abc';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getSortGoods = (goods, sortField, isReversed) => {
+  const preparedList = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedList.sort((a, b) => {
+      switch (sortField) {
+        case sortByLength:
+          return a.length - b.length;
+        case sortByABC:
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    preparedList.reverse();
+  }
+
+  return preparedList;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isSorted, setIsSorted] = useState(false);
+
+  const finalGoods = getSortGoods(goodsFromServer, sortField, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== sortByABC,
+          })}
+          onClick={() => {
+            setSortField(sortByABC);
+            setIsSorted(true);
+            setIsReversed(false);
+          }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**